### PR TITLE
User-configurable stack name in init-odkx-sync-endpoint.py

### PR DESCRIPTION
**Pull request by Ndacyayisenga-droid**
fixes (https://github.com/odk-x/tool-suite-X/issues/194)

cc @linl33 @elmps2018 @wbrunette @e2themillions

### CODE REVIEW

File: init-odkx-sync-endpoint.py [hunk 1]
```diff
@@ -17,12 +17,15 @@ def run_interactive_config():
     env_file_location = os.path.join(os.path.dirname(__file__), "config", "https.env")
 
     try:
-        domain, email = parse_env_file(env_file_location)
-        print("Found configuration at {}".format(env_file_location))
-    except OSError:
-        print("No default https configuration file found at expected path {}. This prevents automatically renewing certs!".format(env_file_location))
-        print("Please check your paths and file permissions, and make sure your config repo is up to date.")
-        exit(1)
+    domain, email, stack_name = parse_env_file(env_file_location)
```
Comment (r0ssing) [inline init-odkx-sync-endpoint.py]: This looks good to me, except I think there is an indentation problem? (if that is fixed, the number of modified lines will also be reduced)

File: init-odkx-sync-endpoint.py [hunk 2]
```diff
@@ -44,7 +55,7 @@ def run_interactive_config():
 
         if default_ldap_pwd != "":
             replaceInFile("ldap.env", r"^\s*LDAP_ADMIN_PASSWORD=.*$", "LDAP_ADMIN_PASSWORD={}".format(default_ldap_pwd))
-            print("Password set to: {}".format(default_ldap_pwd))
+            print("Password set to: {}".format(default_ldap_pwd))
```
Comment (r0ssing) [inline init-odkx-sync-endpoint.py]: Trailing whitespace...

File: init-odkx-sync-endpoint.py [hunk 3]
```diff
@@ -94,9 +106,9 @@ def run_interactive_config():
           --non-interactive".format(email, domain))
 
         print("Attempting to save updated https configuration")
-        write_to_env_file(env_file_location, domain, email)
+        write_to_env_file(env_file_location, domain, email, stack_name)
+    return True
```
Comment (r0ssing) [inline init-odkx-sync-endpoint.py]: I think there is the same indentation issue here??

File: init-odkx-sync-endpoint.py [hunk 4]
```diff
@@ -148,20 +164,22 @@ def run_docker_builds():
...
-        os.system("docker stack deploy -c docker-compose.yml syncldap")
+        os.system("docker stack deploy -c docker-compose.yml {}".format(stack_name))
 
 
 if __name__ == "__main__":
     https = run_interactive_config()
     run_docker_builds()
     run_sync_endpoint_build()
-    deploy_stack(https)
+    deploy_stack(https)
```
Comment (r0ssing) [inline init-odkx-sync-endpoint.py]: Missing newline at end of file

**Review by r0ssing — CHANGES_REQUESTED** [review]
When the small indentation/formatting issues are resolved, I think this looks good? 👍

File: init-odkx-sync-endpoint.py [hunk 5]
```diff
@@ -148,16 +164,18 @@ def run_docker_builds():
 
 
 def run_sync_endpoint_build():
-    os.system("git clone -b master --single-branch --depth=1 https://github.com/odk-x/sync-endpoint ; \
```
Comment (wbrunette) [inline init-odkx-sync-endpoint.py:151]: why would we want to clone more than the single-branch for the automated build? Takes up more space and takes longer

**Review by wbrunette — CHANGES_REQUESTED** [review]

**Pull request closed without merging**